### ISSUE TITLE
Installing nbdime breaks jupyter notebook: Error unpacking user from cookie: Extra data

### ISSUE THREAD

**Issue by lumbric** [jupyter/nbdime#652]
I run a local Jupyter server in a conda environment. If I install nbdime, the Jupyter server won't let me save notebooks anylonger or not start the kernel at all. 

## Related issues and links

- jupyter/notebook#2396 seems to be similar but I think it's not the same issue
- [this Q&A](https://stackoverflow.com/questions/75106093/403-forbidden-error-jupyter-server-on-anaconda) on stackoverflow seems to be very similar
- [a very recent post](https://discourse.jupyter.org/t/notebook-forbidden-403/17844) describes the same error

## How to reproduce

Setup a fresh conda environment and install Jupyter:

```
micromamba create -n test_nbdime_jupyter
micromamba activate test_nbdime_jupyter
micromamba install -c conda-forge jupyter
```

At this point I stored the installed versions, see section below:

```
conda env export --no-builds > env-without_nbdime.yml
```

Run Jupyter server:
```
jupyter notebook
```

Then create the notebook and test it. One can connect to the kernel, execute code and save it.

Now stop the jupyter server and install nbdime:

```
micromamba install -c conda-forge nbdime
```

Export installed versions again:
```
conda env export --no-builds > env-with_nbdime.yml
```

Start the jupyter server again:
```
jupyter notebook
```

Now the notebook cannot connect the kernel or the notebook cannot be saved (both has happened to me, not sure why there was different behavior).

The terminal shows the following error messages, see also full log output below:

```
Error unpacking user from cookie: Extra data: line 1 column 6 (char 5)
Clearing invalid/expired login cookie username-localhost-8888
403 POST /nbdime/api/isgit (127.0.0.1) 9.520000ms referer=http://localhost:8888/notebooks/Untitled.ipynb
```

## Workaround

Uninstall nbdime resolves the issue for me.

## Log output

```
[I 11:39:43.967 NotebookApp] Serving notebooks from local directory: /tmp/test
[I 11:39:43.967 NotebookApp] Jupyter Notebook 6.5.2 is running at:
[I 11:39:43.967 NotebookApp] http://localhost:8888/?token=62a0cafdd37e9<SNIP>
[I 11:39:43.967 NotebookApp]  or http://127.0.0.1:8888/?token=62a0cafdd37<SNIP>
[I 11:39:43.967 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 11:39:43.986 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///home/<SNIP>/.local/share/jupyter/runtime/nbserver-15570-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=62a0cafdd37e98216d5f<SNIP>
     or http://127.0.0.1:8888/?token=62a0cafdd37e98216d5<SNIP>
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[W 11:39:51.558 NotebookApp] Notebook Untitled.ipynb is not trusted
[I 11:39:51.699 NotebookApp] Kernel started: 68dc1ad1-86b3-4ea4-949f-b271f35d2257, name: python3
/opt/miniconda3/envs/test_nbdime_jupyter/lib/python3.11/site-packages/jupyter_server/base/handlers.py:240: UserWarning: The Tornado web application does not have an 'identity_provider' defined in its settings. In future releases of jupyter_server, this will be a required key for all subclasses of `JupyterHandler`. For an example, see the jupyter_server source code for how to add an identity provider to the tornado settings: https://github.com/jupyter-server/jupyter_server/blob/v2.0.0/jupyter_server/serverapp.py#L242
  warnings.warn(
[E 11:39:51.708 NotebookApp] Error unpacking user from cookie: Extra data: line 1 column 6 (char 5)
[W 11:39:51.708 NotebookApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:39:51.710 NotebookApp] 403 POST /nbdime/api/isgit (127.0.0.1) 9.520000ms referer=http://localhost:8888/notebooks/Untitled.ipynb
[W 11:39:51.731 NotebookApp] Forbidden
[W 11:39:51.731 NotebookApp] 403 GET /api/contents/Untitled.ipynb/checkpoints?_=1675939191151 (127.0.0.1) 1.620000ms referer=http://localhost:8888/notebooks/Untitled.ipynb
[W 11:39:51.750 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:39:51.756 NotebookApp] 403 GET /api/kernels/68dc1ad1-8<SNIP>/channels?session_id=a211441f42b0460881cc691010aa98f5 (127.0.0.1) 6.890000ms referer=None
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
[W 11:40:00.730 NotebookApp] Couldn't authenticate WebSocket connection
[W 11:40:00.733 NotebookApp] 403 GET /api/kernels/68dc1ad1-86<SNIP>/channels?session_id=a211441f42b0460881cc<SNIP> (127.0.0.1) 5.570000ms referer=None
[W 11:40:09.161 NotebookApp] Forbidden
[W 11:40:09.162 NotebookApp] 403 GET /api/contents/Untitled.ipynb?content=0&_=1675939088599 (127.0.0.1) 2.330000ms referer=http://localhost:8888/notebooks/Untitled.ipynb
[W 11:40:09.181 NotebookApp] Forbidden
[W 11:40:09.182 NotebookApp] 403 PUT /api/contents/Untitled.ipynb (127.0.0.1) 2.140000ms referer=http://localhost:8888/notebooks/Untitled.ipynb
```

## Package versions used

<details><summary>env-without_nbdime.yml (versions before installing nbdime)</summary>
<pre>
name: test_nbdime_jupyter
channels:
  - conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1
  - _openmp_mutex=4.5
  - alsa-lib=1.2.8
  - anyio=3.6.2
  - argon2-cffi=21.3.0
  - argon2-cffi-bindings=21.2.0
  - asttokens=2.2.1
  - attr=2.5.1
  - attrs=22.2.0
  - backcall=0.2.0
  - backports=1.0
  - backports.functools_lru_cache=1.6.4
  - beautifulsoup4=4.11.2
  - bleach=6.0.0
  - bzip2=1.0.8
  - ca-certificates=2022.12.7
  - cairo=1.16.0
  - cffi=1.15.1
  - comm=0.1.2
  - dbus=1.13.6
  - debugpy=1.6.6
  - decorator=5.1.1
  - defusedxml=0.7.1
  - entrypoints=0.4
  - executing=1.2.0
  - expat=2.5.0
  - fftw=3.3.10
  - flit-core=3.8.0
  - font-ttf-dejavu-sans-mono=2.37
  - font-ttf-inconsolata=3.000
  - font-ttf-source-code-pro=2.038
  - font-ttf-ubuntu=0.83
  - fontconfig=2.14.2
  - fonts-conda-ecosystem=1
  - fonts-conda-forge=1
  - freetype=2.12.1
  - gettext=0.21.1
  - glib=2.74.1
  - glib-tools=2.74.1
  - graphite2=1.3.13
  - gst-plugins-base=1.22.0
  - gstreamer=1.22.0
  - gstreamer-orc=0.4.33
  - harfbuzz=6.0.0
  - icu=70.1
  - idna=3.4
  - importlib-metadata=6.0.0
  - importlib_metadata=6.0.0
  - importlib_resources=5.10.2
  - ipykernel=6.21.1
  - ipython=8.9.0
  - ipython_genutils=0.2.0
  - ipywidgets=8.0.4
  - jack=1.9.22
  - jedi=0.18.2
  - jinja2=3.1.2
  - jpeg=9e
  - jsonschema=4.17.3
  - jupyter=1.0.0
  - jupyter_client=8.0.2
  - jupyter_console=6.4.4
  - jupyter_core=5.2.0
  - jupyter_events=0.6.3
  - jupyter_server=2.2.1
  - jupyter_server_terminals=0.4.4
  - jupyterlab_pygments=0.2.2
  - jupyterlab_widgets=3.0.5
  - keyutils=1.6.1
  - krb5=1.20.1
  - lame=3.100
  - ld_impl_linux-64=2.40
  - libcap=2.66
  - libclang=15.0.7
  - libclang13=15.0.7
  - libcups=2.3.3
  - libdb=6.2.32
  - libedit=3.1.20191231
  - libevent=2.1.10
  - libffi=3.4.2
  - libflac=1.4.2
  - libgcc-ng=12.2.0
  - libgcrypt=1.10.1
  - libgfortran-ng=12.2.0
  - libgfortran5=12.2.0
  - libglib=2.74.1
  - libgomp=12.2.0
  - libgpg-error=1.46
  - libiconv=1.17
  - libllvm15=15.0.7
  - libnsl=2.0.0
  - libogg=1.3.4
  - libopus=1.3.1
  - libpng=1.6.39
  - libpq=15.1
  - libsndfile=1.2.0
  - libsodium=1.0.18
  - libsqlite=3.40.0
  - libstdcxx-ng=12.2.0
  - libsystemd0=252
  - libtool=2.4.7
  - libudev1=252
  - libuuid=2.32.1
  - libvorbis=1.3.7
  - libxcb=1.13
  - libxkbcommon=1.0.3
  - libxml2=2.10.3
  - libzlib=1.2.13
  - lz4-c=1.9.4
  - markupsafe=2.1.2
  - matplotlib-inline=0.1.6
  - mistune=2.0.5
  - mpg123=1.31.2
  - mysql-common=8.0.32
  - mysql-libs=8.0.32
  - nbclassic=0.5.1
  - nbclient=0.7.2
  - nbconvert=7.2.9
  - nbconvert-core=7.2.9
  - nbconvert-pandoc=7.2.9
  - nbformat=5.7.3
  - ncurses=6.3
  - nest-asyncio=1.5.6
  - notebook=6.5.2
  - notebook-shim=0.2.2
  - nspr=4.35
  - nss=3.82
  - openssl=3.0.8
  - packaging=23.0
  - pandoc=2.19.2
  - pandocfilters=1.5.0
  - parso=0.8.3
  - pcre2=10.40
  - pexpect=4.8.0
  - pickleshare=0.7.5
  - pip=23.0
  - pixman=0.40.0
  - pkgutil-resolve-name=1.3.10
  - platformdirs=3.0.0
  - ply=3.11
  - prometheus_client=0.16.0
  - prompt-toolkit=3.0.36
  - prompt_toolkit=3.0.36
  - psutil=5.9.4
  - pthread-stubs=0.4
  - ptyprocess=0.7.0
  - pulseaudio=16.1
  - pure_eval=0.2.2
  - pycparser=2.21
  - pygments=2.14.0
  - pyqt=5.15.7
  - pyqt5-sip=12.11.0
  - pyrsistent=0.19.3
  - python=3.11.0
  - python-dateutil=2.8.2
  - python-fastjsonschema=2.16.2
  - python-json-logger=2.0.4
  - python_abi=3.11
  - pyyaml=6.0
  - pyzmq=25.0.0
  - qt-main=5.15.8
  - qtconsole=5.4.0
  - qtconsole-base=5.4.0
  - qtpy=2.3.0
  - readline=8.1.2
  - rfc3339-validator=0.1.4
  - rfc3986-validator=0.1.1
  - send2trash=1.8.0
  - setuptools=67.1.0
  - sip=6.7.7
  - six=1.16.0
  - sniffio=1.3.0
  - soupsieve=2.3.2.post1
  - stack_data=0.6.2
  - terminado=0.17.1
  - tinycss2=1.2.1
  - tk=8.6.12
  - toml=0.10.2
  - tornado=6.2
  - traitlets=5.9.0
  - typing-extensions=4.4.0
  - typing_extensions=4.4.0
  - tzdata=2022g
  - wcwidth=0.2.6
  - webencodings=0.5.1
  - websocket-client=1.5.1
  - wheel=0.38.4
  - widgetsnbextension=4.0.5
  - xcb-util=0.4.0
  - xcb-util-image=0.4.0
  - xcb-util-keysyms=0.4.0
  - xcb-util-renderutil=0.3.9
  - xcb-util-wm=0.4.1
  - xorg-kbproto=1.0.7
  - xorg-libice=1.0.10
  - xorg-libsm=1.2.3
  - xorg-libx11=1.7.2
  - xorg-libxau=1.0.9
  - xorg-libxdmcp=1.1.3
  - xorg-libxext=1.3.4
  - xorg-libxrender=0.9.10
  - xorg-renderproto=0.11.1
  - xorg-xextproto=7.3.0
  - xorg-xproto=7.0.31
  - xz=5.2.6
  - yaml=0.2.5
  - zeromq=4.3.4
  - zipp=3.12.1
  - zlib=1.2.13
  - zstd=1.5.2
prefix: /opt/miniconda3/envs/test_nbdime_jupyter
</pre>
</details>

<details><summary>env-with_nbdime.yml  (versions before installing nbdime)</summary>
<pre>
name: test_nbdime_jupyter
channels:
  - conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1
  - _openmp_mutex=4.5
  - alsa-lib=1.2.8
  - anyio=3.6.2
  - argon2-cffi=21.3.0
  - argon2-cffi-bindings=21.2.0
  - asttokens=2.2.1
  - attr=2.5.1
  - attrs=22.2.0
  - backcall=0.2.0
  - backports=1.0
  - backports.functools_lru_cache=1.6.4
  - beautifulsoup4=4.11.2
  - bleach=6.0.0
  - brotlipy=0.7.0
  - bzip2=1.0.8
  - ca-certificates=2022.12.7
  - cairo=1.16.0
  - certifi=2022.12.7
  - cffi=1.15.1
  - charset-normalizer=2.1.1
  - colorama=0.4.6
  - comm=0.1.2
  - cryptography=39.0.1
  - dbus=1.13.6
  - debugpy=1.6.6
  - decorator=5.1.1
  - defusedxml=0.7.1
  - entrypoints=0.4
  - executing=1.2.0
  - expat=2.5.0
  - fftw=3.3.10
  - flit-core=3.8.0
  - font-ttf-dejavu-sans-mono=2.37
  - font-ttf-inconsolata=3.000
  - font-ttf-source-code-pro=2.038
  - font-ttf-ubuntu=0.83
  - fontconfig=2.14.2
  - fonts-conda-ecosystem=1
  - fonts-conda-forge=1
  - freetype=2.12.1
  - gettext=0.21.1
  - gitdb=4.0.10
  - gitpython=3.1.30
  - glib=2.74.1
  - glib-tools=2.74.1
  - graphite2=1.3.13
  - gst-plugins-base=1.22.0
  - gstreamer=1.22.0
  - gstreamer-orc=0.4.33
  - harfbuzz=6.0.0
  - icu=70.1
  - idna=3.4
  - importlib-metadata=6.0.0
  - importlib_metadata=6.0.0
  - importlib_resources=5.10.2
  - ipykernel=6.21.1
  - ipython=8.9.0
  - ipython_genutils=0.2.0
  - ipywidgets=8.0.4
  - jack=1.9.22
  - jedi=0.18.2
  - jinja2=3.1.2
  - jpeg=9e
  - jsonschema=4.17.3
  - jupyter=1.0.0
  - jupyter-server-mathjax=0.2.6
  - jupyter_client=8.0.2
  - jupyter_console=6.4.4
  - jupyter_core=5.2.0
  - jupyter_events=0.6.3
  - jupyter_server=2.2.1
  - jupyter_server_terminals=0.4.4
  - jupyterlab_pygments=0.2.2
  - jupyterlab_widgets=3.0.5
  - keyutils=1.6.1
  - krb5=1.20.1
  - lame=3.100
  - ld_impl_linux-64=2.40
  - libcap=2.66
  - libclang=15.0.7
  - libclang13=15.0.7
  - libcups=2.3.3
  - libdb=6.2.32
  - libedit=3.1.20191231
  - libevent=2.1.10
  - libffi=3.4.2
  - libflac=1.4.2
  - libgcc-ng=12.2.0
  - libgcrypt=1.10.1
  - libgfortran-ng=12.2.0
  - libgfortran5=12.2.0
  - libglib=2.74.1
  - libgomp=12.2.0
  - libgpg-error=1.46
  - libiconv=1.17
  - libllvm15=15.0.7
  - libnsl=2.0.0
  - libogg=1.3.4
  - libopus=1.3.1
  - libpng=1.6.39
  - libpq=15.1
  - libsndfile=1.2.0
  - libsodium=1.0.18
  - libsqlite=3.40.0
  - libstdcxx-ng=12.2.0
  - libsystemd0=252
  - libtool=2.4.7
  - libudev1=252
  - libuuid=2.32.1
  - libvorbis=1.3.7
  - libxcb=1.13
  - libxkbcommon=1.0.3
  - libxml2=2.10.3
  - libzlib=1.2.13
  - lz4-c=1.9.4
  - markupsafe=2.1.2
  - matplotlib-inline=0.1.6
  - mistune=2.0.5
  - mpg123=1.31.2
  - mysql-common=8.0.32
  - mysql-libs=8.0.32
  - nbclassic=0.5.1
  - nbclient=0.7.2
  - nbconvert=7.2.9
  - nbconvert-core=7.2.9
  - nbconvert-pandoc=7.2.9
  - nbdime=3.1.1
  - nbformat=5.7.3
  - ncurses=6.3
  - nest-asyncio=1.5.6
  - notebook=6.5.2
  - notebook-shim=0.2.2
  - nspr=4.35
  - nss=3.82
  - openssl=3.0.8
  - packaging=23.0
  - pandoc=2.19.2
  - pandocfilters=1.5.0
  - parso=0.8.3
  - pcre2=10.40
  - pexpect=4.8.0
  - pickleshare=0.7.5
  - pip=23.0
  - pixman=0.40.0
  - pkgutil-resolve-name=1.3.10
  - platformdirs=3.0.0
  - ply=3.11
  - prometheus_client=0.16.0
  - prompt-toolkit=3.0.36
  - prompt_toolkit=3.0.36
  - psutil=5.9.4
  - pthread-stubs=0.4
  - ptyprocess=0.7.0
  - pulseaudio=16.1
  - pure_eval=0.2.2
  - pycparser=2.21
  - pygments=2.14.0
  - pyopenssl=23.0.0
  - pyqt=5.15.7
  - pyqt5-sip=12.11.0
  - pyrsistent=0.19.3
  - pysocks=1.7.1
  - python=3.11.0
  - python-dateutil=2.8.2
  - python-fastjsonschema=2.16.2
  - python-json-logger=2.0.4
  - python_abi=3.11
  - pyyaml=6.0
  - pyzmq=25.0.0
  - qt-main=5.15.8
  - qtconsole=5.4.0
  - qtconsole-base=5.4.0
  - qtpy=2.3.0
  - readline=8.1.2
  - requests=2.28.2
  - rfc3339-validator=0.1.4
  - rfc3986-validator=0.1.1
  - send2trash=1.8.0
  - setuptools=67.1.0
  - sip=6.7.7
  - six=1.16.0
  - smmap=3.0.5
  - sniffio=1.3.0
  - soupsieve=2.3.2.post1
  - stack_data=0.6.2
  - terminado=0.17.1
  - tinycss2=1.2.1
  - tk=8.6.12
  - toml=0.10.2
  - tornado=6.2
  - traitlets=5.9.0
  - typing-extensions=4.4.0
  - typing_extensions=4.4.0
  - tzdata=2022g
  - urllib3=1.26.14
  - wcwidth=0.2.6
  - webencodings=0.5.1
  - websocket-client=1.5.1
  - wheel=0.38.4
  - widgetsnbextension=4.0.5
  - xcb-util=0.4.0
  - xcb-util-image=0.4.0
  - xcb-util-keysyms=0.4.0
  - xcb-util-renderutil=0.3.9
  - xcb-util-wm=0.4.1
  - xorg-kbproto=1.0.7
  - xorg-libice=1.0.10
  - xorg-libsm=1.2.3
  - xorg-libx11=1.7.2
  - xorg-libxau=1.0.9
  - xorg-libxdmcp=1.1.3
  - xorg-libxext=1.3.4
  - xorg-libxrender=0.9.10
  - xorg-renderproto=0.11.1
  - xorg-xextproto=7.3.0
  - xorg-xproto=7.0.31
  - xz=5.2.6
  - yaml=0.2.5
  - zeromq=4.3.4
  - zipp=3.12.1
  - zlib=1.2.13
  - zstd=1.5.2
prefix: /opt/miniconda3/envs/test_nbdime_jupyter
</pre>
</details>

```diff
--- env-without_nbdime.yml	2023-02-09 13:09:15.733223614 +0100
+++ env-with_nbdime.yml	2023-02-09 13:13:31.443793641 +0100
@@ -17,11 +17,16 @@
   - backports.functools_lru_cache=1.6.4
   - beautifulsoup4=4.11.2
   - bleach=6.0.0
+  - brotlipy=0.7.0
   - bzip2=1.0.8
   - ca-certificates=2022.12.7
   - cairo=1.16.0
+  - certifi=2022.12.7
   - cffi=1.15.1
+  - charset-normalizer=2.1.1
+  - colorama=0.4.6
   - comm=0.1.2
+  - cryptography=39.0.1
   - dbus=1.13.6
   - debugpy=1.6.6
   - decorator=5.1.1
@@ -40,6 +45,8 @@
   - fonts-conda-forge=1
   - freetype=2.12.1
   - gettext=0.21.1
+  - gitdb=4.0.10
+  - gitpython=3.1.30
   - glib=2.74.1
   - glib-tools=2.74.1
   - graphite2=1.3.13
@@ -62,6 +69,7 @@
   - jpeg=9e
   - jsonschema=4.17.3
   - jupyter=1.0.0
+  - jupyter-server-mathjax=0.2.6
   - jupyter_client=8.0.2
   - jupyter_console=6.4.4
   - jupyter_core=5.2.0
@@ -122,6 +130,7 @@
   - nbconvert=7.2.9
   - nbconvert-core=7.2.9
   - nbconvert-pandoc=7.2.9
+  - nbdime=3.1.1
   - nbformat=5.7.3
   - ncurses=6.3
   - nest-asyncio=1.5.6
@@ -152,9 +161,11 @@
   - pure_eval=0.2.2
   - pycparser=2.21
   - pygments=2.14.0
+  - pyopenssl=23.0.0
   - pyqt=5.15.7
   - pyqt5-sip=12.11.0
   - pyrsistent=0.19.3
+  - pysocks=1.7.1
   - python=3.11.0
   - python-dateutil=2.8.2
   - python-fastjsonschema=2.16.2
@@ -167,12 +178,14 @@
   - qtconsole-base=5.4.0
   - qtpy=2.3.0
   - readline=8.1.2
+  - requests=2.28.2
   - rfc3339-validator=0.1.4
   - rfc3986-validator=0.1.1
   - send2trash=1.8.0
   - setuptools=67.1.0
   - sip=6.7.7
   - six=1.16.0
+  - smmap=3.0.5
   - sniffio=1.3.0
   - soupsieve=2.3.2.post1
   - stack_data=0.6.2
@@ -185,6 +198,7 @@
   - typing-extensions=4.4.0
   - typing_extensions=4.4.0
   - tzdata=2022g
+  - urllib3=1.26.14
   - wcwidth=0.2.6
   - webencodings=0.5.1
   - websocket-client=1.5.1
```

I used micromamba v1.1.0 for installing the packages, but I guess this should not be relevant.

**Comment by bollwyvl** [comment]
This may be due to some dangling user-level configuration from different versions of jupyter tools at different major versions.

As a test, try running with e.g. 

```
HOME=/tmp/foo jupyter-notebook --no-browser --debug
```

...and launch a new private browsing window manually with the link provided.

If that clears it up, and you don't mind losing local jupyter settings and state, you can then:

```bash
rm -rf ~/.jupyter ~/.local/share/jupyter
```

(and close all your browsers, clear caches, etc).

In the end, though, this probably isn't an `nbdime` issue.

**Comment by xaverm** [comment]
After my last `conda update --all` I got **exactly same issue**. 

It is an `[E 21:39:29.400 NotebookApp] Error unpacking user from cookie: Extra data: line 1 column 4 (char 3)` that one can see, starting the `jupyter-notebook`.

After that error ... "rien ne va plus": kernels cannot be started properly anymore and jupyter notebooks therefore are broken.

I tried many things suggested on the net in the context of not being able to properly connect to kernels. These involved: using other browsers, clearing browser caches, relaxing cookie add-on restrictions, downgrading various packages ( jupyter_server, jupyter_console, nbclassic, websocket-client, pyzmq, tornado). None made any difference.

Then I stumbled across this issue report: **Removing nbdime fixed the issue for me**.
Not only can I start/restart kernels again, but also the cookie unpacking error is gone.

I have zero understanding, as to why the nbdime package, which has not been part of any update for ~ 2 months on my system introduces such a disastrous issue into the environment after upgrading seemingly unrelated packages. None of the console output from where I started jupyter-notebook in order to see some meaningful messages, gave any indication that the error might be due to nbdime. If there is any info I could offer, I'll do.

@lumbric: Thx!
